File: python/ccxt/test/test_calculate_fee.py
# -*- coding: utf-8 -*-

# THIS IS A MOCKUP FILE IN PROGRESS

# import argparse
import os
import sys
# import json
# import time

# ------------------------------------------------------------------------------

root = os.path.dirname(os.path.abspath(__file__))
sys.path.append(root)

# ------------------------------------------------------------------------------

import ccxt  # noqa: E402

# ------------------------------------------------------------------------------

taker = 0.0025
maker = 0.0010
price = 100.00
amount = 10.00

market = {
    'id': 'foobar',
    'symbol': 'FOO/BAR',
    'base': 'FOO',
    'quote': 'BAR',
    'taker': taker,
    'maker': maker,
    'precision': {
        'price': 8,
        'amount': 8,
    }
}

exchange = ccxt.Exchange({
    'id': 'mock',
    'markets': {'FOO/BAR': market},
})

exchange.calculate_fee(market['symbol'], 'limit', 'sell', amount, price, 'taker', {})

# {'rate': {'quote': 0.0025, 'base': 0.0}, 'cost': {'quote': 2.5, 'base': 0.0}}

exchange.calculate_fee(market['symbol'], 'limit', 'sell', amount, price, 'maker', {})

# {'rate': {'quote': 0.001, 'base': 0.0}, 'cost': {'quote': 1.0, 'base': 0.0}}

File: ts/src/test/base/language_specific/test.calculateFee.ts
// @ts-nocheck
// todo: per https://github.com/ttodua/ccxt/blob/428f5b50da50b7401caa5ac452538fb0f6641af4/ts/src/test/base/test.calculateFee.ts

import assert, { strictEqual, deepEqual } from 'assert';
import ccxt, { Exchange, functions } from '../../../../ccxt.js';

const { index, aggregate, unCamelCase } = functions;

const equal = strictEqual;

function testCalculateFee () {
    const price  = 100.00;
    const amount = 10.00;
    const taker  = 0.0025;
    const maker  = 0.0010;
    const fees   = { taker, maker };
    const market = {
        'id':     'foobar',
        'symbol': 'FOO/BAR',
        'base':   'FOO',
        'quote':  'BAR',
        'taker':   taker,
        'maker':   maker,
        'spot': true,
        'precision': {
            'amount': 0.00000001,
            'price': 0.00000001,
        },
    };

    const exchange = new Exchange ({
        'id': 'mock',
        'markets': {
            'FOO/BAR': market,
        },
    });

    Object.keys (fees).forEach ((takerOrMaker) => {

        const result = exchange.calculateFee (market['symbol'], 'limit', 'sell', amount, price, takerOrMaker, {});

        deepEqual (result, {
            'type': takerOrMaker,
            'currency': 'BAR',
            'rate': fees[takerOrMaker],
            'cost': fees[takerOrMaker] * amount * price,
        });
    });
}

export default testCalculateFee;
